main.tf

```diff
@@ -86,6 +86,59 @@ resource "aws_s3_bucket" "default" {
       uri         = grant.value.uri
     }
   }
+
+  dynamic "replication_configuration" {
+    for_each = var.s3_replication_enabled ? [1] : []
+
+    content {
+      role = aws_iam_role.replication[0].arn
+
+      dynamic "rules" {
+        for_each = var.replication_rules == null ? [] : var.replication_rules
+
+        content {
+          id       = rules.value.id
+          priority = try(rules.value.priority, 0)
+          prefix   = try(rules.value.prefix, null)
+          status   = try(rules.value.status, null)
+
+          destination {
+            bucket             = var.s3_replica_bucket_arn
+            storage_class      = try(rules.value.destination.storage_class, "STANDARD")
```

Comment:
what I noticed is that if the first expression of try evaluates to null, the null is used w/o throwing any errors and w/o using the second param.
The above will prob work in any case (b/c `STANDARD` is the default anyway) , but just something to note.

Reply:
hey @aknysh , I gave this a quick try and the issue is that there is no default for the attribute (not even 'null', since this is one piece of a larger object). It throws this error if we don't use the `try`...

```
Error: Unsupported attribute

 107:             storage_class      = rules.value.destination.storage_class
    |----------------
    | rules.value is object with 3 attributes

This object does not have an attribute named "destination".
```

I'm going to mark this one as resolved for now unless there's something else you think I should try.